common/predictive-text/worker/model-compositor.ts

```diff
@@ -9,12 +9,36 @@ class ModelCompositor {
   predict(transformDistribution: Transform | Distribution<Transform>, context: Context): Suggestion[] {
     let suggestionDistribution: Distribution<Suggestion> = [];
 
+    // Assumption:  Duplicated 'displayAs' properties indicate duplicated Suggestions.
+    // When true, we can use an associated array to de-duplicate everything.
+    let suggestionDistribMap: {[key: string]: ProbabilityMass<Suggestion>} = {};
+
     if(!(transformDistribution instanceof Array)) {
       transformDistribution = [ {sample: transformDistribution, p: 1.0} ];
     }
 
+    // Find the transform for the actual keypress.
+    let inputTransform = transformDistribution.sort(function(a, b) {
+      return b.p - a.p;
+    })[0].sample;
+
+    // Only allow new-word suggestions if space was the most likely keypress.
+    let allowSpace = inputTransform.insert == " " || inputTransform.insert == "\n";
```

Comment:
Should this be extended to other, non-ASCII whitespace?

Reply:
I'd say to other, non-ASCII wordbreak characters.  Not sure on the best way to handle that at the moment, personally, though that may be since I'm not the one who implemented the default word breaker.

Reply:
Maybe I don't understand the intent of this code. It's checking for a space or newline because it assumes that...? 

If you're looking specifically for whitespace, then use all characters with Unicode general property Z*:

```javascript
/[\u0020\u00a0\u1680\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u2028\u2029\u202f\u205f\u3000]/
```

These DO NOT include CR, LF, and Tab. So you could use this regex to include them:

```javascript
/[\u0009\u000A\u000D\u0020\u00a0\u1680\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u2028\u2029\u202f\u205f\u3000]/
```

We can address this in a future PR, though! 😅